src/components/FeedbackItem.tsx

```diff
@@ -1,58 +1,116 @@
+	id: string;
 	name: string;
 	timeAgo: string;
 	feedback: string | null;
+	userId: string | null;
+	currentUserId: string | null;
+	onEditComment: (commentId: string | null, content: string) => void;
 }
 
 const FeedbackItem: React.FC<FeedbackItemProps> = ({
+	id,
 	name,
 	timeAgo,
 	feedback,
+	userId,
+	currentUserId,
+	onEditComment,
 }) => {
+	const [isEditing, setIsEditing] = useState(false);
+	const [editedFeedback, setEditedFeedback] = useState(feedback ?? "");
+
+	const handleEditClick = () => {
+		setIsEditing(true);
+	};
+
+	const handleCancelClick = () => {
+		setIsEditing(false);
+		setEditedFeedback(feedback ?? "");
+	};
+
+	const handleSaveClick = async () => {
+		console.log(
+			"handleSaveClick called with userId:",
+			userId,
+			"and editedFeedback:",
+			editedFeedback,
+		);
+
+		if (userId) {
+			onEditComment(id, editedFeedback);
+			setIsEditing(false);
+		} else {
+			console.log("userId is null, not calling onEditComment");
+		}
+	};
+
 	return (
-		<Box sx={{ width: "100%", marginBottom: "var(--spacing-small)" }}>
+		<Box
+			sx={{
+				display: "flex",
+				flexDirection: "column",
+				gap: "4px",
+				padding: "var(--spacing-small)",
+				backgroundColor: "var(--color-background-paper)",
+				borderRadius: "8px",
+				border: "1px solid var(--color-border)",
+			}}
+		>
 			<Box
 				sx={{
 					display: "flex",
 					justifyContent: "space-between",
 					alignItems: "center",
 				}}
 			>
-				<Typography
-					sx={{
-						fontWeight: "var(--font-weight-bold)",
-						fontSize: "18px",
-						fontFamily: "var(--font-family-outfit)",
-					}}
-				>
-					{name}
+				<Typography sx={{ fontWeight: 600 }}>{name}</Typography>
+				<Typography variant="body2" color="textSecondary">
+					{timeAgo}
 				</Typography>
-				<Typography
+			</Box>
+			{isEditing ? (
+				<Box sx={{ display: "flex", alignItems: "center", gap: "8px" }}>
+					<TextField
+						value={editedFeedback}
+						onChange={(e) => setEditedFeedback(e.target.value)}
+						fullWidth
+						size="small"
+					/>
+					<IconButton onClick={handleSaveClick} size="small">
+						<SaveIcon />
+					</IconButton>
+					<IconButton onClick={handleCancelClick} size="small">
+						<CancelIcon />
+					</IconButton>
+				</Box>
+			) : (
+				<Box
 					sx={{
-						color: "#C1C1C1",
-						fontSize: "14px",
-						fontFamily: "var(--font-family-outfit)",
-						whiteSpace: "nowrap",
+						display: "flex",
+						justifyContent: "space-between",
+						alignItems: "center",
 					}}
 				>
-					{timeAgo}
-				</Typography>
-			</Box>
-
-			<Typography
-				sx={{
-					color: "#807F7F",
-					fontSize: "16px",
-					fontFamily: "var(--font-family-outfit)",
-					marginTop: "4px",
-				}}
-			>
-				{feedback}
-			</Typography>
+					<Typography>{feedback}</Typography>
+					{userId === currentUserId && (
+						<IconButton onClick={handleEditClick} size="small">
+							<EditIcon />
+						</IconButton>
+					)}
+				</Box>
+			)}
```

Comment:
Hi Matviy this looks good!
I have tested the PR and ensured the following:
1. Users cannot edit comments that are not there's
2. The Edited comment is persisted even after a reload 
3. The Commenting functionality still works. 


Should be good to be merged.

Reply:
great! thank you!